compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -10287,6 +10356,98 @@ private STNode parseLetClause(boolean isRhsExpr) {
         return STNodeFactory.createLetClauseNode(letKeyword, letVarDeclarations);
     }
 
+    /**
+     * Parse order by clause.
+     * <p>
+     * <code>order-by-clause := order by expression (ascending | descending)? (expression (ascending | descending)?)*
+     * </code>
+     *
+     * @return Parsed node
+     */
+    private STNode parseOrderByClause(boolean isRhsExpr) {
+        STToken nextToken = peek();
+        if (nextToken.kind != SyntaxKind.ORDER_KEYWORD) {
+            return STNodeFactory.createEmptyNode();
+        }
```

Comment:
I think we can avoid this check. If first token is not `ORDER_KEYWORD` it will be handled inside parseOrderKeyword()